pkg/kubelet/kubelet.go

```diff
@@ -1313,6 +1323,19 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Cont
 	if err != nil {
 		return nil, err
 	}
+
+	// If required, create special mounts to order cAdvisor to gather custom metrics.
+	if kl.enableCustomMetrics {
+		customMetricsMount, err := custommetrics.MakeCustomMetricsMountIfRequired(
+			pod, container, kl.getPodContainerDir(pod.UID, container.Name), kl.maxCustomMetricsPerPod)
+		if err != nil {
```

Comment:
If a user define more custom metrics than the allowed maximum number, the pod will be in the pending phase and SyncPod will retry indefinitely. Is that the intended behavior?